### PR TITLE
Implement job priorities in daily report

### PR DESCRIPTION
**Pull request by Crola1702**
# Description

With a lot of new issues been created in the buildfarm for both ROS and Gazebo projects, a way to compare issues by importance or 'priority' is a good enhancement to get rid of most issues.

## Changes

* Add `get_known_issue_errors.sql` to get the error names and jobs from a github issue link
* Add `job_priorities.csv` defined from [Buildfarm Issue Triage](https://docs.google.com/spreadsheets/d/1gX7Hw_aVsI4U05i2-NxCY66a4CmRy1Yyb6dokK5QtMg/edit?gid=0#gid=0) spreadsheet
* Calculate job priority based on the second method defined in #93 
* Show known issues sorted by priority in the report

### CODE REVIEW

**Comment by Crola1702** [comment]
Generated reports:
* JSON Report: https://github.com/osrf/buildfarm-tools/actions/runs/11407891668/artifacts/2075271735
* Markdown Report: https://github.com/osrf/buildfarm-tools/actions/runs/11407891668/artifacts/2075271824

File: database/scripts/lib/buildfarm_tools.rb [hunk 1]
```diff
@@ -128,13 +131,49 @@ def self.jobs_last_success_date(older_than_days: 0)
...
+      end
+    end
+
+    unless sort_by.nil?
+      out.sort_by! { |r| -r.first['priority'] }
+    end
     out
   end
 
+  def self.issue_priority(issue_link)
```
Comment (Blast545) [inline database/scripts/lib/buildfarm_tools.rb:151]: `calculate_issue_priority` instead?

File: database/scripts/lib/buildfarm_tools.rb [hunk 2]
```diff
@@ -128,13 +131,49 @@ def self.jobs_last_success_date(older_than_days: 0)
...
+        job_priority = JOB_PRIORITIES[job]
+        job_priority = job_priority*1.5 if flaky_ratio == 1
+        
+        error_score_jobs[job] = [] if error_score_jobs[job].nil?
+        error_score_jobs[job] << (job_priority*flaky_ratio)
+      end
+    end
+    
+    # Get only maximum score for each job
+    error_score_jobs.each_value.map {|e| e.max}.sum.round(3)
```
Comment (Blast545) [inline database/scripts/lib/buildfarm_tools.rb:174]: Why do we need the maximum here? If you are iterating over the error flakiness in each of the jobs, isn't the total priority of them the complete sum?

**Review by Blast545 — APPROVED** [review]
overall lgtm, minor nits, can you post a preview of the report in .md format before merging for the sake of taking a look?

I would advise not merging it until we come back from ROSCon, just in case to avoid having the scripts broken while we are not around.

File: database/scripts/lib/buildfarm_tools.rb (hunk 2 above)
Comment (Crola1702) [reply database/scripts/lib/buildfarm_tools.rb:174]: The idea behind this is to reduce the "amount of priority" that we get when an error appears in a lot of jobs.

**Pull request merged**